cpp/CMakeLists.txt

```diff
@@ -14,713 +14,567 @@
+    # TODO: Is this always a valid assumption?
+    if(NOT TARGET GTest::gtest)
+        add_library(GTest::gtest ALIAS gtest)
+        add_library(GTest::gmock ALIAS gmock)
+        add_library(GTest::gtest_main ALIAS gtest_main)
+        add_library(GTest::gmock_main ALIAS gmock_main)
+    endif()
+endif()
 
 ###################################################################################################
-# - add gtest -------------------------------------------------------------------------------------
+# - add tests -------------------------------------------------------------------------------------
 
-if(BUILD_TESTS)
+if(CUDF_BUILD_TESTS)
+    # include CTest module -- automatically calls enable_testing()
     include(CTest)
-    include(ConfigureGoogleTest)
-
-    if(GTEST_FOUND)
-        message(STATUS "Google C++ Testing Framework (Google Test) found in ${GTEST_ROOT}")
-        include_directories(${GTEST_INCLUDE_DIR})
-        add_subdirectory(${CMAKE_SOURCE_DIR}/tests)
-    else()
-        message(AUTHOR_WARNING "Google C++ Testing Framework (Google Test) not found: automated tests are disabled.")
-    endif(GTEST_FOUND)
-endif(BUILD_TESTS)
-
-message(STATUS "CUDF_TEST_LIST set to: ${CUDF_TEST_LIST}")
+    add_subdirectory(tests)
+endif()
 
 ###################################################################################################
-# - add google benchmark --------------------------------------------------------------------------
-
-if(BUILD_BENCHMARKS)
-
-  include(ConfigureGoogleBenchmark)
-
-  if(GBENCH_FOUND)
-    message(STATUS "Google C++ Benchmarking Framework (Google Benchmark) found in ${GBENCH_ROOT}")
-    include_directories(${GBENCH_INCLUDE_DIR})
-    add_subdirectory(${CMAKE_SOURCE_DIR}/benchmarks)
-  else()
-    message(AUTHOR_WARNING "Google C++ Benchmarking Framework (Google Benchmark) not found: automated tests are disabled.")
-  endif(GBENCH_FOUND)
-
-endif(BUILD_BENCHMARKS)
-
-message(STATUS "BENCHMARK_LIST set to: ${BENCHMARK_LIST}")
+# - add benchmarks --------------------------------------------------------------------------------
+
+if(CUDF_BUILD_BENCHMARKS)
+    # Find or install GoogleBench
+    CPMFindPackage(NAME benchmark
+        VERSION         1.5.2
+        GIT_REPOSITORY  https://github.com/google/benchmark.git
+        GIT_TAG         v1.5.2
+        GIT_SHALLOW     TRUE
+        OPTIONS         "BENCHMARK_ENABLE_TESTING OFF"
+                        "BENCHMARK_ENABLE_INSTALL OFF")
+    add_subdirectory(benchmarks)
+endif()
 
 ###################################################################################################
-# - include paths ---------------------------------------------------------------------------------
+# - library targets -------------------------------------------------------------------------------
 
-include_directories("${THRUST_INCLUDE_DIR}"
-                    "${JITIFY_INCLUDE_DIR}"
-                    "${LIBCUDACXX_INCLUDE_DIR}")
+add_library(cudf
```

Comment:
Is there any way to make this so we don't have to keep adding files to this list?

Reply:
we can move to globbing if we are happy with the trade-offs. In general it has the downside that cmake has to re-run for every incremental build

Reply:
That doesn't sound worth it so I guess lets just continue as is for now.